src/Workspaces/Core/Portable/SymbolKey/SymbolKey.ErrorTypeSymbolKey.cs

```diff
@@ -13,9 +15,23 @@ private static class ErrorTypeSymbolKey
             public static void Create(INamedTypeSymbol symbol, SymbolKeyWriter visitor)
             {
                 visitor.WriteString(symbol.Name);
-                visitor.WriteSymbolKey(symbol.ContainingSymbol as INamespaceOrTypeSymbol);
```

Comment:
I'm not quite getting why this had to be changed here -- the symbol key written out for the namespace of the error symbol would still have been a valid symbol key, right? Would this also have been fixed if we just had the resolving of the namespace just call CreateErrorNamespaceSymbol if it can't find the original namespace?

Reply:
> the symbol key written out for the namespace of the error symbol would still have been a valid symbol key, right? Would this also have been fixed if we just had the resolving of the namespace just call CreateErrorNamespaceSymbol if it can't find the original namespace?

1. i went down that path and felt *very* oogy about that.  i.e. i didn't like the idea of just blindly resolving as an ErrorNamespace if we couldnt' find it again.  I want that behavior to be explicit only if that's what went in.  
2. i also went down a path of actually exposing an `IsMissing/IsError` bit on INamespaceSymbol so that we could just recurse to the parent and store that bit and *then* rehydrate the ErrorNamespace on the other end.  Indeed, my early commits show that.  However, it was much larger in scope, and as i thought about it, i realized it wasn't necessary, so i went with the simpelr appraoch.